ydb/library/yql/minikql/mkql_program_builder.cpp

```diff
@@ -1611,6 +1611,25 @@ TRuntimeNode TProgramBuilder::BlockCoalesce(TRuntimeNode first, TRuntimeNode sec
     return TRuntimeNode(callableBuilder.Build(), false);
 }
 
+TRuntimeNode TProgramBuilder::BlockExists(TRuntimeNode data) {
+    auto dataType = AS_TYPE(TBlockType, data.GetStaticType());
+    auto dataItemType = dataType->GetItemType();
+
+    if (dataItemType->IsVoid()) {
+        return NewDataLiteral(false);
```

Comment:
output type should be Block, this optimization has already been done for scalar Exists, please remove

Reply:
[Removed](https://github.com/ydb-platform/ydb/pull/3400/files#diff-dc2f75d5908ddf8694d8a7b07b45ffb6c4efd8d7bf93912bb3efddca5e74e168R1614-R1622).